internal/providers/aws/aws.go

```diff
@@ -72,8 +73,21 @@ func NewFetcher(l *log.Logger) (resource.Fetcher, error) {
 
 // Init prepares the fetcher for this platform
 func Init(f *resource.Fetcher) error {
-	// Determine the partition and region this instance is in
-	regionHint, err := ec2metadata.New(f.AWSSession).Region()
+	var regionHint string
+	// During the fetch stage we might be running before the networking is
+	// fully ready. Add a dumb retry (up to 30 seconds worth of retrying).
+	// 30 seconds was chosen entirely arbitrarily, normal HTTP calls would
+	// use a similar method with an infinite timeout but that shouldn't
+	// be done here unless the error detection is updated.
```

Comment:
It's not great to fall back to a partially-broken state after 30 seconds.  What would need to change about the error detection to support an infinite timeout?

Reply:
We could check for [this error](https://github.com/coreos/ignition/blob/master/vendor/github.com/aws/aws-sdk-go/aws/ec2metadata/api.go#L199) and then assume if we get either the other two failure states ([1](https://github.com/coreos/ignition/blob/c4631ca035cb6f76f1b612d286cb9c2fcde04331/vendor/github.com/aws/aws-sdk-go/aws/ec2metadata/api.go#L140-L142) or [2](https://github.com/coreos/ignition/blob/c4631ca035cb6f76f1b612d286cb9c2fcde04331/vendor/github.com/aws/aws-sdk-go/aws/ec2metadata/api.go#L147-L149)) it is likely network / platform related and will be resolved with time but I'm not confident in that assumption (especially for case 2).

Reply:
As discussed OOB switched to using a `FetchToBuffer` call temporarily (with a comment explaining alternative solutions) until we can spend more time digging into the failure states of this call to make sure we would only exit for legitimate failures (extending the timeout to an arbitrarily large number wouldn't work as since this runs each stage it could potentially block a boot from coming up in a reasonable amount of time).

Reply:
It'd be good to put the explanation from https://github.com/coreos/ignition/pull/1155#discussion_r551765111 into the comment block.